### ISSUE TITLE
azurerm_kubernetes_cluster: Terraform Plan shows AKS Cluster is being destroyed and recreated when trying to update the ssh_key for an existing AKS Cluster 

### ISSUE THREAD

**Issue by umersyed** [hashicorp/terraform-provider-azurerm#28128]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.3.10

### AzureRM Provider Version

4.10.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
Update / re-generate new pair of ssh_key for an existing AKS Cluster
```

### Debug Output/Panic Output

```shell
Terraform Plan:

The AKS Cluster is being destroyed / replaced
 # module.aks.azurerm_kubernetes_cluster.aks must be replaced
-/+ resource "azurerm_kubernetes_cluster" "aks" {
...
...
...
 ~ linux_profile {
            # (1 unchanged attribute hidden)

          ~ ssh_key {
              ~ key_data = <<-EOT # forces replacement
...
...
...
}
```

### Expected Behaviour

The Terraform Plan should show only the "ssh_key" for the AKS Cluster should be updated . The AKS Cluster should not be destroyed / replaced.

### Actual Behaviour

The Terraform Plan is showing that the AKS Cluster is being destroyed / replaced.

### Steps to Reproduce

Try to update / regenerate new pair of  ssh_key for an existing AKS Cluster

**Comment by stephybun** [comment]
@ms-henglu you added `ForceNew` back onto `key_data` in the `ssh_key` block in #27105 which was part of the move from stable to preview. Was/Is updating the `ssh_key` for AKS clusters a preview feature?

**Comment by ms-henglu** [comment]
Yes, but let me double check with service team to see if this feature is supported in a stable api-version.

**Comment by umersyed** [comment]
@ms-henglu @stephybun 
1. Do you know when this feature will be released and will go to GA ? 
2. Any link you can share where I can track this feature release ?

**Comment by ms-henglu** [comment]
Hi @umersyed - No concrete ETA yet. A workaround is using `azapi` provider which allows you to use any api-versions including the preview ones.

**Comment by umersyed** [comment]
Hi,

Any update on this issue ?

**Comment by cailyoung** [comment]
@ms-henglu any update here?
